flask_sqlalchemy/__init__.py

```diff
@@ -372,6 +373,47 @@ def next_num(self):
             return None
         return self.page + 1
 
+    @staticmethod
+    def _serialize_item(value):
+        """Handles specific JSON serialization of different datatypes"""
+        if isinstance(value, bool):
+            if value:
+                return 'true'
+            return 'false'
+        if isinstance(value, int) or isinstance(value, float):
+            return value
+        if isinstance(value, str):
+            return value
+        if isinstance(value, dt.datetime):
+            return value.isoformat()
+        if isinstance(value, list):
+            return [Pagination._serialize_item(i) for i in value]
+        if isinstance(value, dict):
+            return {Pagination._serialize_item(k): Pagination._serialize_item(v)
+                    for k, v in value.items()}
+        msg = "Unable to serialize {}".format(value)
+        raise NotImplementedError(msg)
+
+    @staticmethod
+    def _serialize_row(row_proxy):
+        """Converts each row proxy into a dict and serializes the values"""
+        return {k: Pagination._serialize_item(v)
```

Comment:
maybe dumb again but: should this yield (k, v) pairs rather than building up the entire dict before returning?